### DISCUSSION TITLE
how i can read files from a multipart request?

### DISCUSSION THREAD

**Discussion by m1guer** [tokio-rs/axum#2734] [Q&A] [answered]
### Summary

Hi friends, i'm trying read a file and the content inside then from a multipart form and i'm stuck, dont know how to proceed.
i'm using this example [here](https://docs.rs/axum/latest/axum/extract/struct.Multipart.html).

### axum version

0.7.5

**Comment by wangziling** [comment, marked as answer]
Hi Miguel~

Here's a demo code in this blog: https://blog.hiibolt.com/rust/axum/tokio/server/2024/01/10/MultipartRustServer.html#result

You can follow it to get the file `data`(bytes) and file `name`.
The blog shows that the file is in a multipart form data and named with `fileupload`.

**Reply by jplatte** [reply]
If the http service is meant to accept untrusted user input, measures should likely also be taken to avoid overwriting existing files, and even more importantly to make sure that the file name does not contain a `/` or `\` (normalizing it is generally a good idea).

Finally, `std::io` should not be used within `async fn`. Use `tokio`s `io` (and `fs`) modules instead. 

See the example [here](https://docs.rs/tokio/latest/tokio/fs/struct.OpenOptions.html#method.create_new) for robust overwrite protection.

**Comment by m1guer** [comment]
i solved my error. what is relationated with the file size.

**Reply by askor** [reply]
How did you solve it?
